sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/ContainerRegistryBlobClient.cs

```diff
@@ -405,44 +395,51 @@ private async Task<ChunkedUploadResult> UploadInChunksInternal(string location,
             }
 
             // Otherwise, upload in multiple chunks.
-            byte[] buffer = new byte[chunkSize];
-            int chunkCount = 0;
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(chunkSize);
+            long chunkCount = 0;
             long blobLength = 0;
             using SHA256 sha256 = SHA256.Create();
 
-            // Read first chunk into buffer.
-            int bytesRead = async ?
-                await stream.ReadAsync(buffer, 0, chunkSize, cancellationToken).ConfigureAwait(false) :
-                stream.Read(buffer, 0, chunkSize);
-
-            while (bytesRead > 0)
+            try
             {
-                var contentRange = GetContentRange(chunkCount * chunkSize, bytesRead);
-                location = uploadChunkResult?.Headers.Location ?? location;
-
-                // Incrementally compute hash for digest.
-                sha256.TransformBlock(buffer, 0, bytesRead, buffer, 0);
+                // Read first chunk into buffer.
+                int bytesRead = async ?
+                    await stream.ReadAsync(buffer, 0, chunkSize, cancellationToken).ConfigureAwait(false) :
+                    stream.Read(buffer, 0, chunkSize);
 
-                using (Stream chunk = new MemoryStream(buffer, 0, bytesRead))
+                while (bytesRead > 0)
                 {
-                    uploadChunkResult = async ?
-                        await _blobRestClient.UploadChunkAsync(location, chunk, contentRange, bytesRead.ToString(CultureInfo.InvariantCulture), cancellationToken).ConfigureAwait(false) :
-                        _blobRestClient.UploadChunk(location, chunk, contentRange, bytesRead.ToString(CultureInfo.InvariantCulture), cancellationToken);
+                    var contentRange = GetContentRange(chunkCount * chunkSize, bytesRead);
+                    location = uploadChunkResult?.Headers.Location ?? location;
+
+                    // Incrementally compute hash for digest.
+                    sha256.TransformBlock(buffer, 0, bytesRead, buffer, 0);
+
+                    using (Stream chunk = new MemoryStream(buffer, 0, bytesRead))
```

Comment:
nit: is there a way to pass buffer (without reading it into intermediary `chunk` stream)? Please ignore my comment if it's all generated code and it can't be easily done.

Reply:
I like the idea!  It is calling into generated code, so it would not be easy to accomplish, unfortunately.